### PR TITLE
Flatten event type and expose surplus

### PR DESCRIPTION
**Pull request by gmbronco**
This reverts commit 0be3a6e26dbdef9bcfe7daaca999ff05ce08838b.

### CODE REVIEW

File: modules/pool/pool.gql [hunk 1]
```diff
@@ -1503,37 +1503,8 @@ interface GqlPoolEvent {
...
-    sender: String!
-    userAddress: String!
-    poolId: String!
-    timestamp: Int!
-    valueUSD: Float!
-    tokenIn: GqlPoolEventAmount!
-    tokenOut: GqlPoolEventAmount!
+    tokens: [GqlPoolEventAmount!]
+    tokenIn: GqlPoolEventAmount
+    tokenOut: GqlPoolEventAmount
```
Comment (franzns) [inline modules/pool/pool.gql]: is `tokens` for add/remove and `tokenIn`/`tokenOut` for swap?

Comment (gmbronco) [reply modules/pool/pool.gql]: yes, like you said

File: modules/pool/pool.gql [hunk 2]
```diff
@@ -1502,7 +1502,26 @@ interface GqlPoolEvent {
...
+    """
+    tokenIn: GqlPoolEventAmount
+    """
+    Swap output token
+    """
+    tokenOut: GqlPoolEventAmount
+    """
+    Surplus generate by CowAmm swap
+    """
+    surplus: GqlPoolEventAmount
```
Comment (franzns) [inline modules/pool/pool.gql]: should we also add `fee: GqlPoolEventAmount` to make it complete?

File: modules/pool/pool.gql [hunk 3]
```diff
@@ -1502,7 +1502,26 @@ interface GqlPoolEvent {
...
+    """
+    Swap input token
+    """
+    tokenIn: GqlPoolEventAmount
+    """
+    Swap output token
+    """
+    tokenOut: GqlPoolEventAmount
+    """
+    Surplus generate by CowAmm swap
```
Comment (gmbronco) [inline modules/pool/pool.gql]:
```suggestion
    Surplus generated by CowAmm swap
```

File: modules/pool/pool.gql (hunk 2 above)
Comment (gmbronco) [reply modules/pool/pool.gql]: good idea

File: modules/controllers/event-query-controller.ts [hunk 4]
```diff
@@ -126,7 +139,11 @@ export function EventsQueryController(tracer?: any) {
             });
 
             const results = dbEvents.map((event) =>
-                event.type === 'SWAP' ? parseSwap(event as SwapEvent) : parseJoinExit(event as JoinExitEvent),
+                event.type === 'SWAP' && event.protocolVersion === 1
```
Comment (franzns) [inline modules/controllers/event-query-controller.ts]: I'm a bit scared about these protocolVersion tests as they will break if we ever get another custom amm based on v1

Comment (gmbronco) [reply modules/controllers/event-query-controller.ts]: true, it makes sense to check just for surplus

File: modules/pool/pool.gql (hunk 2 above)
Comment (gmbronco) [reply modules/pool/pool.gql]: added

**Pull request merged**